CHANGELOG.md

```diff
@@ -1,3 +1,9 @@
+## 0.24.0 / 2023-05-16
+
+* [CHANGE] Make Proxy Connect Headers consistent with Prometheus #1008
```

Comment:
Q: do we want to mention CVEs that were fixed in https://github.com/prometheus/blackbox_exporter/pull/1072? I see that we mentioned it in 0.23.0 [release](https://github.com/prometheus/blackbox_exporter/blob/master/CHANGELOG.md#0230--2022-12-02)